### PR TITLE
Hail query service

### PR DESCRIPTION
**Pull request by hanars**
Adds a lightweight http service for querying hail. Would like any feedback around the docker image/ resource allocation in kubernetes, and also I'm curious how best to initialize hail -currently I just call hail when I want to use it and it does whatever the default is

@tpoterba for some reason I can't add you as a reviewer but feedback would be helpful

### CODE REVIEW

**Comment by hanars** [comment]
Closing this out so it can be reviewed at this point, while allowing me to continue working on the main branch

File: deploy/kubectl_helpers/utils/hail-search.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  template:
+    metadata:
+      labels:
+        name: hail-search
+        app.kubernetes.io/name: hail-search
+        deployment: prototype
+    spec:
+      volumes:
+      - name: hail-datasets
+        gcePersistentDisk:
```
Comment (sjahl) [inline deploy/kubectl_helpers/utils/hail-search.yaml:39]: I think this `gcePersistentDisk` volume driver is actually deprecated -- you'll want to use a persistentVolume and claim: https://cloud.google.com/kubernetes-engine/docs/how-to/persistent-volumes/readonlymany-disks#pre-existing-persistent-disk

**Review by sjahl — COMMENTED** [review]
Couple suggestions/notes inline. I think this is fine for iterating on the prototype. To get prod-ready, I think we'll want to do some work incorporating the k8s manifest into one of the seqr helm charts, and automate the docker build.

File: deploy/kubectl_helpers/build_hail_search_image.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
+
+docker build . -f deploy/docker/hail_search/Dockerfile -t gcr.io/seqr-project/seqr:prototype-hail-search
+docker push gcr.io/seqr-project/seqr:prototype-hail-search
```
[the remark this answers is not in the document]
Comment (hanars) [reply deploy/kubectl_helpers/build_hail_search_image.sh:4]: > I'd probably opt to use a separate image/repo here,

That makes sense. Do we have to do anything special to set that up? Or just change the image name?

>  before we go to production with this, we'll want to set up a cloudbuild 

Agreed! I added this mostly so I wouldn't make a dumb mistake while iterating/testing. We will definitely want this in cloudbuild and and we will want a helm chart for the kubernetes config

**Pull request closed without merging**